src/controllers/dynakube/activegate/internal/proxy/reconciler_test.go

```
@@ -0,0 +1,177 @@
+		assert.NoError(t, err)
+	})
+	t.Run(`ensure proxy secret deleted`, func(t *testing.T) {
+		var testClient = fake.NewClientBuilder().WithObjects(&corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      capability.BuildProxySecretName(),
+				Namespace: testNamespace,
+			},
+		}).Build()
+		r := newTestReconciler(testClient)
+		update, err := r.Reconcile()
+
+		var proxySecret corev1.Secret
+		var clientError = r.client.Get(context.TODO(), client.ObjectKey{Name: capability.BuildProxySecretName(), Namespace: testNamespace}, &proxySecret)
+
+		assert.Empty(t, proxySecret)
+		assert.True(t, update)
+		assert.True(t, k8serrors.IsNotFound(clientError))
+		assert.NoError(t, err)
+	})
+}
+
+func TestReconcileProxyValue(t *testing.T) {
+	t.Run(`reconcile proxy Value`, func(t *testing.T) {
+		var proxyValue = buildProxyUrl(proxyUsername, proxyPassword, proxyHost, proxyPort)
+		r := newTestReconciler(fake.NewClientBuilder().Build())
+		r.dynakube.Spec.Proxy = &dynatracev1beta1.DynaKubeProxy{Value: proxyValue}
+		update, err := r.Reconcile()
+
+		var proxySecret corev1.Secret
+		r.client.Get(context.TODO(), client.ObjectKey{Name: capability.BuildProxySecretName(), Namespace: testNamespace}, &proxySecret)
+
+		assert.Equal(t, []byte(proxyPassword), proxySecret.Data[proxyPasswordField])
+		assert.Equal(t, []byte(proxyPort), proxySecret.Data[proxyPortField])
+		assert.Equal(t, []byte(proxyHost), proxySecret.Data[proxyHostField])
+		assert.Equal(t, []byte(proxyUsername), proxySecret.Data[proxyUsernameField])
+		assert.True(t, update)
+		assert.NoError(t, err)
+	})
+	t.Run(`reconcile empty proxy Value`, func(t *testing.T) {
+		r := newTestReconciler(fake.NewClientBuilder().Build())
+		r.dynakube.Spec.Proxy = &dynatracev1beta1.DynaKubeProxy{Value: ""}
+		update, err := r.Reconcile()
+
+		var proxySecret corev1.Secret
+		r.client.Get(context.TODO(), client.ObjectKey{Name: capability.BuildProxySecretName(), Namespace: testNamespace}, &proxySecret)
```

same as bevore, do not implicitly ignore an error